### PR TITLE
Auth feature layout

### PR DESCRIPTION
**Pull request by dbrosio3**
## Links:

* [Trello Card](https://trello.com/c/5QOv8lHL/2-2-as-a-user-i-should-be-able-to-sign-up-to-create-an-account)
* [Mockups](https://www.figma.com/file/79uiALRVOHTMEHyqShdHnA/Black-Market?node-id=1%3A1457)

## What & Why:

((Apologies for the __huge__ file count, It's mostly due to styling)) 

**Included in this PR:**
- Setup [Chakra-ui](https://chakra-ui.com/), customized theme
- Created Login dummy form, applied styling
- Idem Registration

**_Not_ included in this PR:**
- i18n
- form validations 
- real registration / login functionality   

image

image

## Risks, Mitigation & Rollback:

- [ ] Safe to Revert *check this box if this PR can be reverted without incident*

### CODE REVIEW

File: src/features/auth/components/Login/LoginFooter.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import { PATHS } from '@/routes/paths';
+import { FullWidthButton } from '@/styles/theme/components/Buttons';
+
+export const LoginFooter = () => {
+  const navigate = useNavigate();
+
+  return (
+    <>
+      <Text>{`Don't have an account?`}</Text>
```
Comment (flomosq) [inline src/features/auth/components/Login/LoginFooter.tsx:14]:
```suggestion
      <Text>Don't have an account?</Text>
```

File: src/features/auth/components/Register/RegisterForm.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                  variant="unstyled"
+                  aria-label="Toggle password visibility"
+                  icon={icon}
+                />
+              </InputRightElement>
+            )}
+          </InputGroup>
+        </>
+      ))}
+      <FullWidthButton colorScheme="secondary" disabled onClick={handleRegister}>
```
Comment (flomosq) [inline src/features/auth/components/Register/RegisterForm.tsx:83]: Why is it disabled? Are you changing it in a future PR?

File: src/features/auth/components/Register/RegistrationFooter.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { Container, Link, Text } from '@chakra-ui/react';
+import { Link as RouterLink } from 'react-router-dom';
+
+import { PATHS } from '@/routes/paths';
+
+export const RegistrationFooter = () => (
+  <Container w="110%" mt="1rem" px="0" mx="-0.5rem">
+    <Text align="center">
+      By signing up, you accept our{` `}
+      <Link as={RouterLink} to={`/${PATHS.TERMS_AND_CONDITIONS}`} colorScheme="lightblue">
```
Comment (flomosq) [inline src/features/auth/components/Register/RegistrationFooter.tsx:12]: what about adding the `/` to the path? So you could just do 
```
to={PATHS.TERMS_AND_CONDITIONS}
```

File: src/features/auth/components/Register/RegistrationFooter.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        Conditions
+      </Link>
+      , the{' '}
```
Comment (flomosq) [inline src/features/auth/components/Register/RegistrationFooter.tsx:15]: would be nice if once you add i18n you could remove these added spaces

File: src/styles/index.css [hunk 5]
```diff
@@ -0,0 +1,9 @@
+html {
+  overflow: hidden !important;
+}
+
+body {
+  height: 100vh;
+  width: 100vw;
+  background-color: #00031a !important;
```
Comment (flomosq) [inline src/styles/index.css:8]: is the important needed?

Comment (dbrosio3) [reply src/styles/index.css:8]: Yes, I found the `ChakraProvider` resets/overrides some styles

File: src/features/auth/components/Register/RegistrationFooter.tsx (hunk 3 above)
Comment (dbrosio3) [reply src/features/auth/components/Register/RegistrationFooter.tsx:12]: That works for using it in a `<Link>`. Unfortunately not in a `<Route>`

```
// path has to be without the `/`
<Route path={PATHS.REGISTER} element={<Register />} />
```

File: src/features/auth/components/Register/RegisterForm.tsx (hunk 2 above)
Comment (dbrosio3) [reply src/features/auth/components/Register/RegisterForm.tsx:83]: It's so it matches the mockups. I'll add the disabled logic along with the validations.

File: src/features/auth/components/Login/LoginFooter.tsx (hunk 1 above)
Comment (dbrosio3) [reply src/features/auth/components/Login/LoginFooter.tsx:14]: It's to get rid of this:

>`'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`.eslint [react/no-unescaped-entities](https://github.com/yannickcr/eslint-plugin-react/tree/master/docs/rules/no-unescaped-entities.md)

Can we leave it like this for now? It's going to get properly fixed when I add internationalization.

Comment (flomosq) [reply src/features/auth/components/Login/LoginFooter.tsx:14]: Oh yeah, sorry I didn't see the `'`

**Review by flomosq — APPROVED** [review]

**Review by dariokozicki — APPROVED** [review]
LGTM! 🚀

**Pull request merged**